pyquil/operator_estimation.py

```diff
@@ -799,17 +850,169 @@ def measure_observables(qc: QuantumComputer, tomo_experiment: TomographyExperime
+                    calibr_prog += _one_q_pauli_prep(label=op, index=0, qubit=q)
+                # 4.2 Measure the out operator in this state
+                for q, op in setting.out_operator.operations_as_set():
+                    calibr_prog += _local_pauli_eig_meas(op, q)
+                calibr_shots = n_shots
+                calibr_results = qc.run_and_measure(calibr_prog, calibr_shots)
+                # 4.3 Obtain statistics from the measurement process
+                obs_calibr_mean, obs_calibr_var = _stats_from_measurements(calibr_results, setting, calibr_shots)
+                # 4.4 Calibrate the readout results
+                corrected_mean = obs_mean / obs_calibr_mean
+                corrected_var = ratio_variance(obs_mean, obs_var, obs_calibr_mean, obs_calibr_var)
+
+                yield ExperimentResult(
+                    setting=setting,
+                    expectation=obs_mean.item(),
+                    stddev=np.sqrt(obs_var).item(),
+                    total_counts=n_shots,
+                    calibration_expectation=obs_calibr_mean.item(),
+                    calibration_stddev=np.sqrt(obs_calibr_var).item(),
+                    calibration_counts=calibr_shots,
+                    corrected_expectation=corrected_mean.item(),
+                    corrected_stddev=np.sqrt(corrected_var).item(),
+                )
+
+            else:
+                # No calibration
+                obs_calibr_mean = None
+                obs_calibr_var = None
+                calibr_shots = None
+                corrected_mean = None
+                corrected_var = None
+
+                yield ExperimentResult(
+                    setting=setting,
+                    expectation=obs_mean.item(),
+                    stddev=np.sqrt(obs_var).item(),
+                    total_counts=n_shots,
+                )
+
+
+def _ops_strs_symmetrize(qubits: List[int]) -> List[str]:
+    """
+    :param qubits: list specifying the qubits whose readout errors we wish to symmetrize
+    :return: list with the operation strings necessary for exhaustive symmetrization
+    """
+    ops_strings = []
+    for prod in itertools.product(['I', 'X'], repeat=len(qubits)):
+        ops_strings.append(''.join(prod))
+    return ops_strings
+
+
+def _ops_str_to_prog(ops_str: str, qubits: List[int]) -> Program:
+    """
+    :param ops_str: string specifying the operation to be carried out on `qubits`
+    :param qubits: list specifying the qubits to be carried operations on
+    :return: Program with the operations specified in `ops_str` on the qubits specified in `qubits`
+    """
+    assert len(ops_str) == len(qubits), "Mismatch of qubits and operations"
+    prog = Program()
+    for i, op_ch in enumerate(ops_str):
+        if op_ch == 'I':
+            continue
+        elif op_ch == 'X':
+            prog += Program(X(qubits[i]))
+        else:
+            raise ValueError("ops_strings_ should only consist of 'I's and/or 'X's")
```

Comment:
consider using `List[bool]` or a numpy array of dtype bool to avoid messing with strings with this restriction

Reply:
Using `Tuple[bool]` now